### PR TITLE
Add options to specify authorized networks for slurm-cloudsql

### PR DESCRIPTION
**Pull request by andybubu**
This PR adds the options to specify authorized network range for slurm-cloudsql

### CODE REVIEW

File: community/modules/database/slurm-cloudsql-federation/main.tf [hunk 1]
```diff
@@ -49,6 +49,16 @@ resource "google_sql_database_instance" "instance" {
       ipv4_enabled                                  = false
       private_network                               = var.network_id
       enable_private_path_for_google_cloud_services = true
+
+      dynamic "authorized_networks" {
+        for_each = var.authorized_networks
+        iterator = ip_range
+
+        content {
+          name  = "ip-range-${ip_range.key}"
```
Comment (mr0re1) [inline community/modules/database/slurm-cloudsql-federation/main.tf:58]: AFAIK. `name` is optional and having `ip-range-<some arbitrary small int>` as an identifier have a questionable usability. 
Do you mind removing it or provide more information why it's desired ?

Comment (andybubu) [reply community/modules/database/slurm-cloudsql-federation/main.tf]: Got it, this is fixed.

**Review by mr0re1 — APPROVED** [review]

**Comment by mr0re1** [comment]
Please fix failed pre-commit https://github.com/GoogleCloudPlatform/hpc-toolkit/actions/runs/9316575705/job/25659894122?pr=2631

Otherwise your PR is "good to land"

```
--- a/community/modules/database/slurm-cloudsql-federation/README.md
+++ b/community/modules/database/slurm-cloudsql-federation/README.md
@@ -73,7 +73,7 @@ No modules.
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
-| <a name="input_authorized_networks"></a> [authorized\_networks](#input\_authorized\_networks) | IP address ranges as authorized networks of the Cloud SQL for MySQL instances. | `list(string)` | `[]` | no |
+| <a name="input_authorized_networks"></a> [authorized\_networks](#input\_authorized\_networks) | IP address ranges as authorized networks of the Cloud SQL for MySQL instances | `list(string)` | `[]` | no |
 | <a name="input_database_version"></a> [database\_version](#input\_database\_version) | The version of the database to be created. | `string` | `"MYSQL_5_7"` | no |
 | <a name="input_deletion_protection"></a> [deletion\_protection](#input\_deletion\_protection) | Whether or not to allow Terraform to destroy the instance. | `string` | `false` | no |
 | <a name="input_deployment_name"></a> [deployment\_name](#input\_deployment\_name) | The name of the current deployment | `string` | n/a | yes |
```

**Pull request merged**